### DISCUSSION TITLE
How to close Popup from inside?

### DISCUSSION THREAD

**Discussion by KoushikMVN** [ant-design/ant-design-mobile#4013] [Q&A] [answered]
I have face issue to close popup from inside.

Suppose I have a popup like that 

```
<Popup
  visible={viewFilterList}
  onMaskClick={() => {
  setViewFilterList(false)
  }}
  position='right'
  bodyStyle={{minWidth : '95vw'}}
  >
     <CloseCircleOutlined />
</ Popup>

```
I want to close this PopUp when `CloseCircle` icon is click. How can I do this from inside a `PopUp`? And also when page reload `PopUp` is not closing how can I do that also?

**Comment by awmleer** [comment, marked as answer]
Thanks for opening this question!
But sorry I'm a little bit confused. Could you please make a small demo about it on codesandbox?
https://codesandbox.io/s/antd-mobile-snrxr?file=/package.json

**Reply by KoushikMVN** [reply]
Suppose i have a state like 

```
const [viewFilterList, setViewFilterList] = useState(false);
```
and button like 
```
<span   onClick = { ( ) => {
    setViewFilterList ( true )
}}>Filter </span>
```

When I click the filter `span` than above popup will be shown and popup only close when I click outside of this popup but my question is how can I close a popup from inside. You can see [Ant Design Modal](https://codesandbox.io/s/bl8b6). In this Modal when I click cross icon in upper right corner this modal is closed, [Ant Design Modal Page](https://ant.design/components/modal/). So my question is how can I close a `PopUp` from inside like `Ant Design Modal` .

Did you get my point?

**Reply by awmleer** [reply]
Just `setViewFilterList(false)` when `<CloseCircleOutlined />` get clicked?

**Reply by KoushikMVN** [reply]
Thanks it work. Earlier it is not working so I am little bit of concerted about it. BTW Thanks
